purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/PaywallAnimation.kt

```diff
@@ -0,0 +1,53 @@
+package com.revenuecat.purchases.paywalls.components
+
+import com.revenuecat.purchases.InternalRevenueCatAPI
+import dev.drewhamilton.poko.Poko
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonClassDiscriminator
+
+@Suppress("unused")
+@InternalRevenueCatAPI
+@Poko
+@Serializable
+@SerialName("animation")
+class PaywallAnimation(
+    @get:JvmSynthetic val type: AnimationType,
+    @get:JvmSynthetic
+    @SerialName("ms_delay")
+    val msDelay: Int?,
+    @get:JvmSynthetic
+    @SerialName("ms_duration")
+    val msDuration: Int?,
+) {
+
+    @OptIn(ExperimentalSerializationApi::class)
+    @InternalRevenueCatAPI
+    @Serializable
+    @JsonClassDiscriminator("type")
+    sealed class AnimationType {
+        @Serializable
+        @SerialName("ease_in")
+        object EaseIn : AnimationType()
+
+        @Serializable
+        @SerialName("ease_in_out")
+        object EaseInOut : AnimationType()
+
+        @Serializable
+        @SerialName("ease_out")
+        object EaseOut : AnimationType()
+
+        @Serializable
+        @SerialName("linear")
+        object Linear : AnimationType()
+
+        @Poko
+        @Serializable
+        @SerialName("custom")
+        class Custom(
```

Comment:
Hmm I'm not sure how this will work... Would we want developers to give us an animation depending on the `value`? maybe planned for future PRs? If it's not needed for right now, maybe we could make this an enum with a default value and remove this `Custom` type for now.

Reply:
Yea, the thought is to allow users to inject a function that will return an animation (or if looking at the other file, a transition) that way they get the ability to fully customize this feature, but I'm not handling that now, It in this PR really as a placeholder/conversation starter. 

I can remove custom, but I did want to get your thoughts on this concept